ssl/methods.c

```diff
@@ -118,26 +118,7 @@
 
 static const SSL_METHOD *tls1_get_method(int ver)
 {
-    if (ver == TLS_ANY_VERSION)
-        return TLS_method();
-#ifndef OPENSSL_NO_TLS1_2
-    if (ver == TLS1_2_VERSION)
-        return tlsv1_2_method();
-#endif
-#ifndef OPENSSL_NO_TLS1_1
-    if (ver == TLS1_1_VERSION)
-        return tlsv1_1_method();
-#endif
-#ifndef OPENSSL_NO_TLS1
-    if (ver == TLS1_VERSION)
-        return tlsv1_method();
-#endif
-#ifndef OPENSSL_NO_SSL3
-    if (ver == SSL3_VERSION)
-        return (sslv3_method());
-    else
-#endif
-    return NULL;
+    return TLS_method();
```

Comment:
I'll defer to OpenSSL, but given how different 1.0.2 and 1.1.0 are, it'll already be a non-trivial backport. I would probably do it as correctly as possible for 1.1.0 and, once that's working, figure out how to do it in 1.0.2. (For instance, you could leave the `get_ssl_method` hook for ABI reasons but remove the call in `SSL_set_session`. The hook would just be unused within the library.)